boomerang/src/lib.rs

```diff
@@ -12,29 +13,34 @@ pub mod nicp_to_icp;
 #[cfg(test)]
 pub mod state_machine;
 
-// "ryjl3-tyaaa-aaaaa-aaaba-cai"
-pub const ICP_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 2, 1, 1]);
-
-// "buwm7-7yaaa-aaaar-qagva-cai"
-#[cfg(not(feature = "test-env"))]
-pub const NICP_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 2, 48, 1, 170, 1, 1]);
-// "rwlgt-iiaaa-aaaaa-aaaaa-cai"
-#[cfg(feature = "test-env")]
-pub const NICP_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 0, 1, 1]);
-
-// "jcmow-hyaaa-aaaaq-aadlq-cai"
-#[cfg(not(feature = "test-env"))]
-pub const WTN_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 2, 0, 0, 215, 1, 1]);
-// "renrk-eyaaa-aaaaa-aaada-cai"
-#[cfg(feature = "test-env")]
-pub const WTN_LEDGER_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 6, 1, 1]);
-
-// "tsbvt-pyaaa-aaaar-qafva-cai"
-#[cfg(not(feature = "test-env"))]
-pub const WATER_NEURON_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 2, 48, 1, 106, 1, 1]);
-// "r7inp-6aaaa-aaaaa-aaabq-cai"
-#[cfg(feature = "test-env")]
-pub const WATER_NEURON_ID: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 3, 1, 1]);
+#[derive(Deserialize, CandidType, Clone)]
+pub struct CanisterIds {
+    pub icp_ledger_id: Principal,
+    pub nicp_ledger_id: Principal,
+    pub wtn_ledger_id: Principal,
+    pub water_neuron_id: Principal,
+}
+
+thread_local! {
+    static __STATE: RefCell<Option<CanisterIds>> = RefCell::default();
```

Comment:
Maybe this should be named `__CANISTER_IDS`?

Reply:
done